tests/RouterTest.php

```diff
@@ -144,15 +144,33 @@ public function testMethodNotAllowedResponses(
     list($responder, $_data) =
       $router->routeMethodAndPath(HttpMethod::HEAD, 'getonly');
     expect($responder)->toBeSame('getonly');
-    expect(() ==> $router->routeMethodAndPath(HttpMethod::GET, 'headonly'))->toThrow(
-      MethodNotAllowedException::class,
-    );
-    expect(() ==> $router->routeMethodAndPath(HttpMethod::HEAD, 'postonly'))->toThrow(
-      MethodNotAllowedException::class,
-    );
-    expect(() ==> $router->routeMethodAndPath(HttpMethod::GET, 'postonly'))->toThrow(
-      MethodNotAllowedException::class,
-    );
+
+    try {
+      $router->routeMethodAndPath(HttpMethod::GET, 'headonly');
+      static::fail(
+        Str\format('Failed asserting that %s was thrown.', MethodNotAllowedException::class)
+      );
+    } catch(MethodNotAllowedException $e) {
+      expect($e->getAllowedMethods())->toContain(HttpMethod::HEAD);
```

Comment:
`toBeSame(keyset[...])` might be better so that we also test that there are no extra unexpected elements returned